mobile/src/main/java/org/openhab/habdroid/ui/widget/WidgetImageView.kt

```diff
@@ -45,377 +45,389 @@ import org.openhab.habdroid.util.getPrefs
 import org.openhab.habdroid.util.isDebugModeEnabled
 import org.openhab.habdroid.util.resolveThemedColor
 
-class WidgetImageView(context: Context, attrs: AttributeSet?) : AppCompatImageView(context, attrs) {
-    private var scope: CoroutineScope? = null
-    private val fallback: Drawable?
-
-    private var skeleton: Skeleton? = null
-    private var originalScaleType: ScaleType? = null
-    private var originalAdjustViewBounds: Boolean = false
-    private val emptyHeightToWidthRatio: Float
-    private val addRandomnessToUrl: Boolean
-    private var imageScalingType = ImageScalingType.NoScaling
-    private var internalLoad: Boolean = false
-    private var lastRequest: HttpImageRequest? = null
-
-    private var refreshInterval: Long = 0
-    private var lastRefreshTimestamp: Long = 0
-    private var refreshJob: Job? = null
-    private var refreshActive = false
-    private var pendingRequest: PendingRequest? = null
-    private var pendingLoadJob: Job? = null
-    private var targetImageSize: Int = 0
+class WidgetImageView(context: Context, attrs: AttributeSet?, private val imageView: InternalImageView) :
+    SkeletonLayout(context, attrs, config = createConfig(context)),
+    WidgetImageViewIntf by imageView
+{
+    constructor(context: Context, attrs: AttributeSet?) : this(
+        context,
+        attrs,
+        InternalImageView(context, attrs)
+    )
 
     init {
-        context.obtainStyledAttributes(attrs, R.styleable.WidgetImageView).apply {
-            fallback = getDrawable(R.styleable.WidgetImageView_fallback)
-            emptyHeightToWidthRatio = getFraction(R.styleable.WidgetImageView_emptyHeightToWidthRatio, 1, 1, 0f)
-            addRandomnessToUrl = getBoolean(R.styleable.WidgetImageView_addRandomnessToUrl, false)
-            val imageScalingType = getInt(R.styleable.WidgetImageView_imageScalingType, 0)
-            if (imageScalingType < ImageScalingType.entries.size) {
-                setImageScalingType(ImageScalingType.entries[imageScalingType])
-            }
-            recycle()
-        }
-        // In some cases it's required to add the skeleton manually to XML.
-        // In these cases, set the parent as skeleton here, otherwise create it in applyProgressDrawable()
-        val parent = parent
-        if (parent is SkeletonLayout) {
-            skeleton = parent
-        }
```

Comment:
BTW, @mueller-ma, not sure whether you're aware of the syntax: `skeleton = parent as? SkeletonLayout` would have been shorter ;-)